### PR TITLE
John/frontend login

### PR DESCRIPTION
**Pull request by jt5301**
Somewhat preliminary pr for any feedback on styling for signup login. Also wrote localstorage functions to store+get+erase token, authaxios for protected routes that would be tested sometime later. The idea would be to send the token as authorization.

Also: right now its set up so that newly registered users should have to log in after creation so that the frontend can receive a token for any auth purposes later. 
Screen Shot 2020-07-02 at 12 51 39 AM

### CODE REVIEW

**Review by grogsy — APPROVED** [review]
Looks good. Hurrah for React hooks!

File: client/src/pages/Signup.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        setErrorMessage('Please fill in all fields')
+        passedFields = false
+      }
+      if (!passedFields) {
+        setOpen(true)
+        return
+      }
+
+      //if frontend validations pass, make server call here to create user
+      const res = await axios.get('welcome', {
```
Comment (powerseed) [inline client/src/pages/Signup.js]: I think sign up should use "post" instead of "get"?

File: client/src/pages/Signup.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+      }
+      if (password.length < 7) {
+        setErrorMessage('Please enter a password longer than 7 characters')
+        passedFields = false
+      }
+      if (email === '' || company === '' || password === '') {
+        setErrorMessage('Please fill in all fields')
+        passedFields = false
+      }
```
Comment (powerseed) [inline client/src/pages/Signup.js:61]: Do we want to check empty **before** validation?

File: client/src/pages/Signup.js (hunk 1 above)
Comment (jt5301) [reply client/src/pages/Signup.js]: Yes it's post, i am adding some minor things and changed + pushed a commit accidentally during testing. Next commit should fix it!

File: client/src/pages/Signup.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            type="password"
+            id="password"
+            onChange={event => { setPassword(event.target.value) }}
+          />
+          <SubmitButton>
+            {loginSignup === '/signup' ? 'Create' : 'Log In'}
+          </SubmitButton>
+        </form>
+      </div>
+      <Snackbar
```
Comment (powerseed) [inline client/src/pages/Signup.js]: The position of this Snackbar is at the right-bottom corner now, maybe it is better to place it above the textfields.

File: client/src/pages/Login.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    try {
+      event.preventDefault()
+      let passedFields = true
+      if (email === '' || password === '') {
+        setErrorMessage('Please fill in all fields')
+        passedFields = false
+      }
+
+      if (!validateEmail(email)) {
+        setErrorMessage('Please enter your email')
```
Comment (powerseed) [inline client/src/pages/Login.js]: I think "_Please enter a valid email_" could be clearer.

File: client/src/pages/Login.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        passedFields = false
+      }
+      if (password.length < 7) {
+        setErrorMessage('Please enter your password')
```
Comment (powerseed) [inline client/src/pages/Login.js]: I think "Please enter a password with at least 7 characters." could be clearer.

File: client/src/pages/Signup.js (hunk 3 above)
Comment (jt5301) [reply client/src/pages/Signup.js]: Something like this?

Screen Shot 2020-07-02 at 7 43 58 PM

File: client/src/pages/Login.js (hunk 4 above)
Comment (jt5301) [reply client/src/pages/Login.js]: will do~

File: client/src/pages/Login.js (hunk 5 above)
Comment (jt5301) [reply client/src/pages/Login.js]: will do~

File: client/src/pages/Signup.js (hunk 3 above)
Comment (powerseed) [reply client/src/pages/Signup.js]: I am not sure. I think some websites put the error display between "Create an account" and the textfields, some others put it between the textfields and the "Create" button. What do you think? @grogsy

Comment (jt5301) [reply client/src/pages/Signup.js]: material ui textfields have an option to display errors at the bottom of the input. If this looks better I should be able to switch over:
Screen Shot 2020-07-02 at 8 00 08 PM

Comment (powerseed) [reply client/src/pages/Signup.js]: Yeah I think this is better~

Comment (jt5301) [reply client/src/pages/Signup.js]: I kind of agree too. Ok I can switch them

**Comment by jt5301** [comment]
Apologies for the delay @aeciorc, I had some trouble writing automatic login after creating an account.

Below is a gif walkthrough of what would happen on 401 (after "creating" account). I added another interceptor to reset state to hide routes if a user is not found on storage. 

Alt Text

**Review by aeciorc — APPROVED** [review]
Looks good for merging. One improvement left is that the style for the signup and login components are duplicated, instead you could create a single style object and use it from both components

**Comment by jt5301** [comment]
Thanks Aecio. I can merge then refactor!

**Pull request merged**